torch/utils/_foreach_utils.py

```diff
@@ -21,6 +22,14 @@
 def _group_tensors_by_device_and_dtype(tensorlistlist: List[List[Tensor]],
                                        with_indices: Optional[bool] = False) -> \
         Dict[Tuple[torch.device, torch.dtype], List[List[Union[Tensor, int]]]]:
+
+    if is_compiling():
+        if with_indices:
+            indices = list(range(len(tensorlistlist[0])))
+            tensorlistlist.append(indices)
```

Comment:
Is this the correct output format? @janeyx99 I think it is but want you to double check

Reply:
Also lemme know if you'd rather have me do this a different way

Reply:
Adding a helper